packit_service/worker/testing_farm.py

```diff
@@ -56,6 +56,20 @@ def tft_api_url(self) -> str:
                 self._tft_api_url += "/"
         return self._tft_api_url
 
+    @property
+    def fmf_url(self):
+        return (
+            self.job_config.metadata.fmf_url
+            or self.project.get_pr(self.metadata.pr_id).source_project.get_web_url()
+        )
+
+    @property
+    def fmf_ref(self):
+        if self.job_config.metadata.fmf_url:
```

Comment:
Why do we check for `fmf_url` and not `fmf_ref`?

Reply:
There is no need to check for `fmf_ref`. If it isn't set, `None` is returned and it shouldn't be a problem.

I check for `fmf_url` because I want to use either (project repo + project ref) or (remote repo + remote ref). I don't want to mix them. 

This way if `fmf_ref` is set, but `fmf_url` isn't, tests are taken from project repository and current branch/ref (`fmf_ref` is ignored).

If those parameters were independent it would lead to problems. Eg. if `fmf_url` was set, `fmf_ref` wasn't set and project ref was used instead, testing farm would receive the remote repo url (`fmf_url`) with invalid ref (project repo commit-sha)

I agree that this may be a bit confusing.